parts/configure-swarmmode-cluster.sh

```diff
@@ -0,0 +1,302 @@
+echo "BASESUBNET: $BASESUBNET"
+echo "AZUREUSER: $AZUREUSER"
+
+###################
+# Common Functions
+###################
+
+ensureAzureNetwork()
+{
+  # ensure the host name is resolvable
+  hostResolveHealthy=1
+  for i in {1..120}; do
+    host $VMNAME
+    if [ $? -eq 0 ]
+    then
+      # hostname has been found continue
+      hostResolveHealthy=0
+      echo "the host name resolves"
+      break
+    fi
+    sleep 1
+  done
+  if [ $hostResolveHealthy -ne 0 ]
+  then
+    echo "host name does not resolve, aborting install"
+    exit 1
+  fi
+
+  # ensure the network works
+  networkHealthy=1
+  for i in {1..12}; do
+    wget -O/dev/null http://bing.com
+    if [ $? -eq 0 ]
+    then
+      # hostname has been found continue
+      networkHealthy=0
+      echo "the network is healthy"
+      break
+    fi
+    sleep 10
+  done
+  if [ $networkHealthy -ne 0 ]
+  then
+    echo "the network is not healthy, aborting install"
+    ifconfig
+    ip a
+    exit 2
+  fi
+  # ensure the host ip can resolve
+  networkHealthy=1
+  for i in {1..120}; do
+    hostname -i
+    if [ $? -eq 0 ]
+    then
+      # hostname has been found continue
+      networkHealthy=0
+      echo "the network is healthy"
+      break
+    fi
+    sleep 1
+  done
+  if [ $networkHealthy -ne 0 ]
+  then
+    echo "the network is not healthy, cannot resolve ip address, aborting install"
+    ifconfig
+    ip a
+    exit 2
+  fi
+}
+ensureAzureNetwork
+HOSTADDR=`hostname -i`
+
+ismaster ()
+{
+  if [ "$MASTERPREFIX" == "$VMPREFIX" ]
+  then
+    return 0
+  else
+    return 1
+  fi
+}
+if ismaster ; then
+  echo "this node is a master"
+fi
+
+isagent()
+{
+  if ismaster ; then
+    return 1
+  else
+    return 0
+  fi
+}
+if isagent ; then
+  echo "this node is an agent"
+fi
+
+MASTER0IPADDR="${BASESUBNET}${MASTERFIRSTADDR}"
+
+######################
+# resolve self in DNS
+######################
+
+echo "$HOSTADDR $VMNAME" | sudo tee -a /etc/hosts
+
+################
+# Install Docker
+################
+
+echo "Installing and configuring Docker"
+
+installDocker()
+{
+  for i in {1..10}; do
+    wget --tries 4 --retry-connrefused --waitretry=15 -qO- https://get.docker.com | sh
+    if [ $? -eq 0 ]
+    then
+      # hostname has been found continue
+      echo "Docker installed successfully"
+      break
+    fi
+    sleep 10
+  done
+}
+time installDocker
+
+sudo usermod -aG docker $AZUREUSER
+
+echo "Updating Docker daemon options"
+
+updateDockerDaemonOptions()
+{
+    sudo mkdir -p /etc/systemd/system/docker.service.d
+    # Start Docker and listen on :2375 (no auth, but in vnet) and
+    # also have it bind to the unix socket at /var/run/docker.sock
+    sudo bash -c 'echo "[Service]
+    ExecStart=
+    ExecStart=/usr/bin/docker daemon -H tcp://0.0.0.0:2375 -H unix:///var/run/docker.sock
+  " > /etc/systemd/system/docker.service.d/override.conf'
+}
+time updateDockerDaemonOptions
+
+echo "Installing Docker Compose"

```

Comment:
(Future of compose yamls seems to be DABs: https://github.com/docker/docker/blob/master/experimental/docker-stacks-and-bundles.md

Mixes in more k8s pod concepts with compose concepts. Apparently it's not quite ready to really fully do everything compose did, but it works with swarm mode apparently.)